test/redis-invalidation.test.js

```diff
@@ -0,0 +1,214 @@
+      invalidate: () => ['foo'],
+      tap
+    })
+    // Cache the follwoing
+    await request({ app, query: '{ get(id: 11) }' })
+    tap.equal(miss, 1)
+    await request({ app, query: '{ get(id: 12) }' })
+    tap.equal(miss, 2)
+    await request({ app, query: '{ search(id: 11) }' })
+    tap.equal(miss, 3)
+    // Request a mutation
+    await request({ app, query: 'mutation { set(id: 11) }' })
+    tap.equal(miss, 3)
+    // 'get:11' should be still in cache
+    await request({ app, query: '{ get(id: 11) }' })
+    tap.equal(miss, 3)
+    await request({ app, query: '{ search(id: 11) }' })
+    tap.equal(miss, 3)
+    await request({ app, query: '{ get(id: 12) }' })
+    tap.equal(miss, 3)
+  })
+
+  test('should invalidate more than one reference at once', async tap => {
+    // Setup Fastify and Mercurius
+    let miss = 0
+    const app = setupServer({
+      onMiss: () => ++miss,
+      tap
+    })
+    // Cache the follwoing
+    await request({ app, query: '{ get(id: 11) }' })
+    tap.equal(miss, 1)
+    await request({ app, query: '{ get(id: 12) }' })
+    tap.equal(miss, 2)
+    await request({ app, query: '{ search(id: 11) }' })
+    tap.equal(miss, 3)
+    // Request a mutation
+    await request({ app, query: 'mutation { set(id: 11) }' })
+    tap.equal(miss, 3)
+    // All 'get' should not be present in cache anymore
+    await request({ app, query: '{ get(id: 11) }' })
+    tap.equal(miss, 4)
+    await request({ app, query: '{ search(id: 11) }' })
+    tap.equal(miss, 4)
+    await request({ app, query: '{ get(id: 12) }' })
+    tap.equal(miss, 5)
+  })
+
+  test('should remove storage keys by references, but not the ones still alive', async tap => {
+    // Setup Fastify and Mercurius
+    let failHit = false
+    const app = setupServer({
+      onHit () {
+        if (failHit) tap.fail()
+      },
+      tap
+    })
+    // Run the request and cache it
+    await request({ app, query: '{ get(id: 11) }' })
+    tap.equal(
+      await redisClient.get((await redisClient.smembers('r:get:11'))[0]),
+      '"get 11"'
+    )
+    await request({ app, query: '{ get(id: 12) }' })
+    tap.equal(
+      await redisClient.get((await redisClient.smembers('r:get:12'))[0]),
+      '"get 12"'
+    )
+    await request({ app, query: '{ search(id: 11) }' })
+    tap.equal(
+      await redisClient.get((await redisClient.smembers('r:search:11'))[0]),
+      '"search 11"'
+    )
+    // Request a mutation, invalidate 'gets'
+    await request({ app, query: 'mutation { set(id: 11) }' })
+    // Check the references of 'searchs', should still be there
+    tap.equal(
+      await redisClient.get((await redisClient.smembers('r:search:11'))[0]),
+      '"search 11"'
+    )
+    // 'get:11' should not be present in cache anymore,
+    failHit = true
+    // should trigger onMiss and not onHit
+    await request({ app, query: '{ get(id: 11) }' })
+  })
+
+  test('should not throw on invalidation error', async tap => {
```

Comment:
minor:  here and elsewhere, the convention is usually to call this variable just t instead of tap. usually tap is what we call the default export of the tap package

Reply:
good spot Simone! @jhonrocha could you fix it?

Reply:
Done!